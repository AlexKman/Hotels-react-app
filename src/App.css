@import url("https://fonts.googleapis.com/css?family=Pontano+Sans");

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 150vh;
  background-color: #393e41;
}

#header {
  font-size: 5vh;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  margin-bottom: 10vh;
  color: #e94f37;
}
#sortByHeader {
  color: #e94f37;
  font-size: 1.3em;
}

#filterHeader {
  color: #e94f37;
  margin-bottom: 1vh;
  font-size: 1.3em;
}
#filterOptions {
  display: flex;
  flex-direction: column;
  color: #f6f7eb;
}
#filterSection {
  color: #e94f37;
  margin-bottom: 1vh;
  font-size: 1.3em;
}
.hotelsList {
  color: #f6f7eb;
  display: inline-grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto;
  font-family: "Pontano sana", sans-serif;
  font-size: 1.3em;
  grid-gap: 2em;
}
#hotel {
  display: flex;
  flex-direction: column;
  text-decoration: none;
  color: black;
  margin: 0.6vh;
  padding: 0.8vh;
  font-size: 2vh;
  background-color: #bcbbbd;
  border-radius: 14px;
  box-shadow: 0.2em 0.2em 3px 1px #26191b;
  width: 20vw;
  color: black;
}
