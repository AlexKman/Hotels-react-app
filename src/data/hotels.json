[
  {
    "name": "Hilton",
    "starRating": 5,
    "facilities": ["car park", "pool"],
    "id": 1
  },
  {
    "name": "Landmark",
    "starRating": 3,
    "facilities": ["car park", "gym"],
    "id": 2
  },
  {
    "name": "Billay",
    "starRating": 3,
    "facilities": ["pool"],
    "id": 3
  },
  {
    "name": "The Standard",
    "starRating": 10,
    "facilities": ["pool", "car park", "snooker"],
    "id": 4
  },
  {
    "name": "Tranquil Sanctuary",
    "starRating": 1,
    "facilities": ["bar", "gym"],
    "id": 5
  },
  {
    "name": "Vacation hotel",
    "starRating": 7,
    "facilities": ["snooker"],
    "id": 6
  }
]
