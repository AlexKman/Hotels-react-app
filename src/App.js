import React, { Component } from "react";
import hotels from "./data/hotels.json";
import "./App.css";
import Checkbox from "./components/checkbox.jsx";

class App extends Component {
  state = {
    hotels: [],
    facilities: [],
    starsOrder: "asc"
  };

  componentDidMount() {
    this.setState({ hotels });
  }

  render() {
    const { hotels } = this.state;
    return (
      <div className="App">
        <header id="header">Hotels-ReactApp</header>
        <header>Sort by star rating!</header>
        <select onChange={this.handleStarsChange}>
          <option value="asc">Asc</option>
          <option value="desc">Desc</option>
        </select>
        <br />
        <header>Add filters!</header>
        <form id="filteroptions">
          <div>
            Car Park
            <Checkbox onClick={this.handleSearch} id="car park" />
          </div>
          <div>
            Gym
            <Checkbox onClick={this.handleSearch} id="gym" />
          </div>
          <div>
            Pool
            <Checkbox onClick={this.handleSearch} id="pool" />
          </div>
        </form>
        <br />
        Filtering:{" "}
        {this.state.facilities.join(", ") || "showing all facilities!"}
        <ul className="hotelslist">
          {this.sortHotelInStarsOrder()
            .filter(hotel =>
              this.checkForFacilities(hotel.facilities, this.state.facilities)
            )
            .map(hotel => (
              <div id="hotel" key={hotel.id}>
                <section>{hotel.name}</section>
                <section>Star rating = {hotel.starRating}</section>
                <section>Facilities = {this.getHotelFacilities(hotel)}</section>
              </div>
            ))}
        </ul>
      </div>
    );
  }
  checkForFacilities = (itemFacilities, specifiedFacilities) => {
    for (let i = 0; i < specifiedFacilities.length; i += 1) {
      if (!itemFacilities.includes(specifiedFacilities[i])) return false;
    }
    return true;
  };
  getHotelFacilities = hotel => {
    return hotel.facilities.join(", ") || "no facilities";
  };
  sortHotelInStarsOrder = () => {
    const hotels = this.state.hotels;
    hotels.sort((hotel1, hotel2) => {
      return this.state.starsOrder === "asc"
        ? hotel1.starRating - hotel2.starRating
        : hotel2.starRating - hotel1.starRating;
    });
    return hotels;
  };
  handleStarsChange = e => {
    this.setState({ starsOrder: e.target.value });
  };

  handleSearch = e => {
    if (!this.state.facilities.includes(e.target.id)) {
      this.setState({ facilities: [...this.state.facilities, e.target.id] });
    } else {
      const facilitiesCopy = [...this.state.facilities];
      const index = facilitiesCopy.indexOf(e.target.id);
      facilitiesCopy.splice(index, 1);
      this.setState({ facilities: facilitiesCopy });
    }
  };
}

export default App;
