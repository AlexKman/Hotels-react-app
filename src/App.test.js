import { mount } from "enzyme";
import React from "react";
import App from "./App";
import hotels from "./data/hotels.json";

describe("checkbox", () => {
  let component;
  let mockFn;
  beforeEach(() => {
    mockFn = jest.fn();
    component = mount(<App />);
  });
  describe("Component", () => {
    it("renders correctly", () => {
      expect(component).toMatchSnapshot();
    });
    it("has a select box with two options", () => {
      expect(component.find("select")).toHaveLength(1);
    });
    it("has a select box with two options", () => {
      expect(component.find("option")).toHaveLength(2);
    });
    it("has three CheckBox's", () => {
      expect(component.find("Checkbox")).toHaveLength(3);
    });
  });
});
