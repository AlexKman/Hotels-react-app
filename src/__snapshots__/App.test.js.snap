// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox Component renders correctly 1`] = `
<App>
  <div
    className="App"
  >
    <header
      id="header"
    >
      Hotels-ReactApp
    </header>
    <header>
      Sort by star rating!
    </header>
    <select
      onChange={[Function]}
    >
      <option
        value="asc"
      >
        Asc
      </option>
      <option
        value="desc"
      >
        Desc
      </option>
    </select>
    <br />
    <header>
      Add filters!
    </header>
    <form
      id="filteroptions"
    >
      <div>
        Car Park
        <Checkbox
          id="car park"
          onClick={[Function]}
        >
          <input
            id="car park"
            onClick={[Function]}
            type="checkbox"
          />
        </Checkbox>
      </div>
      <div>
        Gym
        <Checkbox
          id="gym"
          onClick={[Function]}
        >
          <input
            id="gym"
            onClick={[Function]}
            type="checkbox"
          />
        </Checkbox>
      </div>
      <div>
        Pool
        <Checkbox
          id="pool"
          onClick={[Function]}
        >
          <input
            id="pool"
            onClick={[Function]}
            type="checkbox"
          />
        </Checkbox>
      </div>
    </form>
    <br />
    Filtering:
     
    showing all facilities!
    <ul
      className="hotelslist"
    >
      <div
        id="hotel"
        key="5"
      >
        <section>
          Tranquil Sanctuary
        </section>
        <section>
          Star rating = 
          1
        </section>
        <section>
          Facilities = 
          bar, gym
        </section>
      </div>
      <div
        id="hotel"
        key="2"
      >
        <section>
          Landmark
        </section>
        <section>
          Star rating = 
          3
        </section>
        <section>
          Facilities = 
          car park, gym
        </section>
      </div>
      <div
        id="hotel"
        key="3"
      >
        <section>
          Billay
        </section>
        <section>
          Star rating = 
          3
        </section>
        <section>
          Facilities = 
          pool
        </section>
      </div>
      <div
        id="hotel"
        key="1"
      >
        <section>
          Hilton
        </section>
        <section>
          Star rating = 
          5
        </section>
        <section>
          Facilities = 
          car park, pool
        </section>
      </div>
      <div
        id="hotel"
        key="6"
      >
        <section>
          Vacation hotel
        </section>
        <section>
          Star rating = 
          7
        </section>
        <section>
          Facilities = 
          snooker
        </section>
      </div>
      <div
        id="hotel"
        key="4"
      >
        <section>
          The Standard
        </section>
        <section>
          Star rating = 
          10
        </section>
        <section>
          Facilities = 
          pool, car park, snooker
        </section>
      </div>
    </ul>
  </div>
</App>
`;
